package/MDAnalysis/core/groups.py

```diff
@@ -377,21 +418,7 @@ def __str__(self):
                                                   repr(list(self)[:3])[:-1],
                                                   repr(list(self)[-3:])[1:])
 
-    def __add__(self, other):
-        """Concatenate the Group with another Group or Component of the same
-        level.
-
-        Parameters
-        ----------
-        other : Group or Component
-            Group or Component with `other.level` same as `self.level`
-
-        Returns
-        -------
-        Group
-            Group with elements of `self` and `other` concatenated
-
-        """
+    def _get_other_index(self, other):
```

Comment:
I don't like this method because it does two things at once (both the level check, as well as returned `.ix`).  I do like centralising the level check so we get consistent error messages etc, but I'd prefer something like a decorator:

```python
@only_same_level
def __add__(self, other):
    # at this point, same_level check has happened

@only_same_level
def union(self, other):
    etc
```

Or if not a decorator, we could have a level check function inside each method